### DISCUSSION TITLE
superset api endpoint - /api/v1/dataset/  results limited ?

### DISCUSSION THREAD

**Discussion by Scope0910** [apache/superset#26178] [Q&A / Help]
hey folks,

I have a question. I use the superset api with endpoint "/api/v1/dataset/" to get the ids for the datasets. We have 25 datasets configured but the result getting from api call only return 20 results as json. Is there a config or something else which limit the resultset from api call ? I dont use rison to query the api or something. 

Any ideas ?
Greets

**Comment by fasilu** [comment]
same issue here is there any limit ?

**Comment by dkrat7** [comment]
API returns datasets by pages. There are "page" and "page_size" parameters (default page size is 20). You can set higher values (I think 100 is page_size upper limit)

Check API [specs](https://superset.apache.org/docs/api/)

**Comment by bronz3beard** [comment]
You can try something like this it works for us.

```typescript
    const response = await axios.get(`${SUPERSET_BASE}/api/v1/dataset/`, {
      headers,
      params: {
        page: 0,
        page_size: 300
      },
      paramsSerializer: (params) => `q=${encodeUrl(JSON.stringify(params))}`
    });
    ```

**Comment by mistercrunch** [comment]
I just checked to make sure it's documented, but seems pretty comprehensive [in the API docs](https://superset.apache.org/docs/api)

<img width="1005" alt="Screenshot 2024-05-08 at 12 27 03 PM" src="https://github.com/apache/superset/assets/487433/1cd2d8e6-f405-4ed0-aa9a-08d5abf542e1">

**Comment by mistercrunch** [comment]
Closing as answer is clear in the comments, and discoverable in docs

**Reply by christiaan** [reply]
The default of "20" is not in the docs. Instead the value of "0" implied to me that by default no limit or pagination is applied.
